core/src/test/java/org/infinispan/notifications/cachelistener/cluster/AbstractClusterListenerTest.java

```diff
@@ -661,4 +661,85 @@ public void testAllExpire() throws InterruptedException {
       assertEquals(key, event.getKey());
       assertEquals(expectedValue, event.getValue());
    }
+   
+   @Test
+   public void testSimpleExpirationFilterNotOwner() {
+      testSimpleExpirationFilter(new MagicKey(cache(1, CACHE_NAME), cache(2, CACHE_NAME)));
+   }
+
+   @Test
+   public void testExpirationMetadataFilterNotOwner() {
+      final String keyToFilterOut = "filter-me";
+      testExpirationFilter(keyToFilterOut, 50l, new MagicKey(cache(1, CACHE_NAME), cache(2, CACHE_NAME)), 1000l,
+              new KeyValueFilterAsCacheEventFilter(new LifespanFilter<Object, String>(100)));
+   }
+
+   protected void testSimpleExpirationFilter(Object key) {
+      final String keyToFilterOut = "filter-me";
+      final long commonLifespan = 1000l;
+      testExpirationFilter(keyToFilterOut, commonLifespan, key, commonLifespan, new KeyFilterAsCacheEventFilter<Object>(
+              new CollectionKeyFilter(Collections.singleton(key), true)));
+   }
+
+   protected void testExpirationFilter(Object keyToFilterOut, Long keyToFilterOutLifespan, Object keyToUse, Long keyToUselifespan, CacheEventFilter<? super Object, ? super String> filter) {
+      Cache<Object, String> cache0 = cache(0, CACHE_NAME);
+
+      //put from a non-owner
+      cache0.put(keyToFilterOut, FIRST_VALUE, keyToFilterOutLifespan, TimeUnit.MILLISECONDS);
+
+      ClusterListener clusterListener = listener();
+      cache0.addListener(clusterListener, filter, null);
+
+      ts0.advance(keyToFilterOutLifespan + 1);
+      ts1.advance(keyToFilterOutLifespan + 1);
+      ts2.advance(keyToFilterOutLifespan + 1);
+      
+       // We should not have gotten the message since it was filtered
+      assertEquals(clusterListener.events.size(), 0);
```

Comment:
The event here won't get fired even if it wasn't filtered unless you accessed the key or the reaper ran.  Maybe switch this and line 699.